### PR TITLE
fix: check in Pipefile.lock for test container

### LINKED ISSUE #1815: bug(ci): test container can sometimes not be build
**What happened:**
The [build / container image for tests](https://github.com/gardenlinux/gardenlinux/blob/c25f1695147c38b0a9e5bb6f8b26df96714a5df6/.github/workflows/build.yml#L100) fails sometimes, which blocks nightly and dev pipelines of Garden Linux 

**Debug Logs:**
Example of failed Pipeline: https://github.com/gardenlinux/gardenlinux/actions/runs/6768874994/job/18394147551#step:5:21663

```
pipenv.exceptions.ResolutionFailure: ERROR: Getting requirements
to build wheel exited with 1
✘ Locking Failed!

Traceback (most recent call last):
  File "/opt/venv/bin/pipenv", line 8, in <module>
    sys.exit(cli())
             ^^^^^
  File "/opt/venv/lib/python3.11/site-packages/pipenv/vendor/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/venv/lib/python3.11/site-packages/pipenv/cli/options.py", line 58, in main
    return super().main(*args, **kwargs, windows_expand_args=False)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/venv/lib/python3.11/site-packages/pipenv/vendor/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/opt/venv/lib/python3.11/site-packages/pipenv/vendor/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/venv/lib/python3.11/site-packages/pipenv/vendor/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/venv/lib/python3.11/site-packages/pipenv/vendor/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/venv/lib/python3.11/site-packages/pipenv/vendor/click/decorators.py", line 92, in new_func
    return ctx.invoke(f, obj, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/venv/lib/python3.11/site-packages/pipenv/vendor/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/venv/lib/python3.11/site-packages/pipenv/cli/command.py", line 209, in install
    do_install(
  File "/opt/venv/lib/python3.11/site-packages/pipenv/routines/install.py", line 164, in do_install
    do_init(
  File "/opt/venv/lib/python3.11/site-packages/pipenv/routines/install.py", line 672, in do_init
    do_lock(
  File "/opt/venv/lib/python3.11/site-packages/pipenv/routines/lock.py", line 65, in do_lock
    venv_resolve_deps(
  File "/opt/venv/lib/python3.11/site-packages/pipenv/utils/resolver.py", line 849, in venv_resolve_deps
    c = resolve(cmd, st, project=project)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/venv/lib/python3.11/site-packages/pipenv/utils/resolver.py", line 718, in resolve
    raise RuntimeError("Failed to lock Pipfile.lock!")
RuntimeError: Failed to lock Pipfile.lock!
Error: error building at STEP "RUN pipenv install --system --dev --verbose": error while running runtime: exit status 1
Error: Process completed with exit code 125.
```

### PR DESCRIPTION
**Pull request by Vincinator**
**What this PR does / why we need it**:
* Pipfile.lock will be  updated via Dependabot
* Generating Pipfile.lock dynamically in container fails. This commit removes the dynamic Pipfile.lock generation in the container build step, thus removes the error source that the Pipfile.lock generation can fail during the container test image build step
* Fixes nightly and dev builds 

**Which issue(s) this PR fixes**:
Fixes #1815 

**Special notes for your reviewer**:
Tested locally with 
```
 podman build --squash --tag test --build-arg base="$(./build --print-container-image)" tests 
```

### CODE REVIEW

File: tests/Containerfile [hunk 1]
```diff
@@ -8,7 +8,7 @@ RUN python3 -m venv $VIRTUAL_ENV
 ENV PATH="$VIRTUAL_ENV/bin:$PATH"
 
 RUN pip install pipenv
-COPY Pipfile Pipfile
```
Comment (fwilhe) [inline tests/Containerfile:11]: Is it correct to remove the pipfile here? Not a python expert, but I'd expect both files are needed.

**Comment by fwilhe** [comment]
I think it makes sense to configure depenabot right away. [Here is an example for a python config](https://github.com/paperless-ngx/paperless-ngx/blob/128594584e00794f7d76cdbb23a5fc0ba101712d/.github/dependabot.yml#L22).

I think we need to adjust the path.

[Dependabot docs for reference](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#pip-and-pip-compile)

File: tests/Containerfile (hunk 1 above)
Comment (Vincinator) [reply tests/Containerfile:11]: The Pipefile is used to generate the `Pipfile.lock`. 

The generation of the Pipfile.lock is moved outside of the container, thus `pipenv` must not re-generate a different `Pipfile.lock` during the container build step. 
Therefore I removed the Pipfile, so pipenv does not even try to generate it. 

However, we could add a `--ignore-pipfile` to pipenv.

Comment (Vincinator) [reply tests/Containerfile:11]: Hm.. let me check. if we even have to use `--ignore-pipfile`.

**Comment by Vincinator** [comment]
> I think it makes sense to configure depenabot right away. [Here is an example for a python config](https://github.com/paperless-ngx/paperless-ngx/blob/128594584e00794f7d76cdbb23a5fc0ba101712d/.github/dependabot.yml#L22).
> 
> I think we need to adjust the path.
> 
> [Dependabot docs for reference](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#pip-and-pip-compile)

Yes, will do

File: tests/Containerfile (hunk 1 above)
Comment (Vincinator) [reply tests/Containerfile:11]: Ok, it's either `--ignore-pipfile` or adding the `Pipfile`to the container image. 

Added Pipfile to the container image now

File: .github/dependabot.yml [hunk 2]
```diff
@@ -8,3 +8,11 @@ updates:
       interval: "weekly"
     reviewers:
       - "gardenlinux/garden-linux-maintainers"
+
+  - package-ecosystem: "pip"
+    directory: "/"
```
Comment (fwilhe) [inline .github/dependabot.yml:13]:
```suggestion
    directory: "/tests"
```

To my understanding this should point to the directory where our pipfile is located

Comment (Vincinator) [reply .github/dependabot.yml:13]: Do we need to limit dependabot to check only Pipfile in tests? Why?

Comment (Vincinator) [reply .github/dependabot.yml:13]: Checking if dependabot scans recursively or needs explicit paths

Comment (Vincinator) [reply .github/dependabot.yml]: Yep - you were right

**Review by fwilhe — APPROVED** [review]
lgtm

File: .github/dependabot.yml (hunk 2 above)
Comment (fwilhe) [reply .github/dependabot.yml]: fyi, dependabot docs describe this: https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#directory

**Pull request merged**